File: skeleton/src/me/shkschneider/skeleton/ui/NonSwipeableViewPager.java
package me.shkschneider.skeleton.ui;

import android.content.Context;
import android.support.v4.view.ViewPager;
import android.util.AttributeSet;
import android.view.MotionEvent;

import org.jetbrains.annotations.NotNull;

public class NonSwipeableViewPager extends ViewPager {

    public NonSwipeableViewPager(@NotNull final Context context) {
        super(context);
    }

    public NonSwipeableViewPager(@NotNull final Context context, final AttributeSet attrs) {
        super(context, attrs);
    }

    @Override
    public boolean onInterceptTouchEvent(final MotionEvent event) {
        return false;
    }

    @Override
    public boolean onTouchEvent(final MotionEvent event) {
        return false;
    }

}

File: android/src/main/kotlin/me/shkschneider/skeleton/android/core/ui/widget/NonSwipeableViewPager.kt
package me.shkschneider.skeleton.ui.widget

import android.annotation.SuppressLint
import android.content.Context
import android.util.AttributeSet
import android.view.KeyEvent
import android.view.MotionEvent
import android.view.View
import androidx.viewpager.widget.ViewPager

class NonSwipeableViewPager(
        context: Context,
        attrs: AttributeSet? = null
) : ViewPager(context, attrs) {

    override fun onInterceptTouchEvent(ev: MotionEvent): Boolean {
        return false
    }

    @SuppressLint("ClickableViewAccessibility")
    override fun onTouchEvent(ev: MotionEvent): Boolean {
        return false
    }

    override fun executeKeyEvent(event: KeyEvent): Boolean {
        return false
    }

//    // Locks if we're at the start or end of the pager
//    override fun canScroll(v: View?, checkV: Boolean, dx: Int, x: Int, y: Int): Boolean {
//        return if (v !== this && v is ViewPager) true else super.canScroll(v, checkV, dx, x, y)
//    }

}
